internal/pkg/docker/dockercompose/decompose.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/compose-spec/compose-go/loader"
+	compose "github.com/compose-spec/compose-go/types"
+	"sort"
+)
+
+// DecomposeService parses a Compose YAML file and then converts a single service to a Copilot manifest.
+func DecomposeService(content []byte, svcName string, workingDir string) (*manifest.BackendServiceConfig, IgnoredKeys, error) {
+	config, err := loader.ParseYAML(content)
+	if err != nil {
+		return nil, nil, fmt.Errorf("parse compose yaml: %w", err)
+	}
+
+	service, err := serviceConfig(config, svcName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get service config: %w", err)
+	}
+
+	ignored, err := unsupportedServiceKeys(service, svcName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get unsupported service keys: %w", err)
+	}
+
+	project, err := loader.Load(compose.ConfigDetails{
+		WorkingDir: workingDir,
+		ConfigFiles: []compose.ConfigFile{
+			{
+				Content: content,
+				Config:  config,
+			},
+		},
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("load Compose project: %w", err)
+	}
+
+	svcConfig, err := project.GetService(svcName)
+	if err != nil {
+		// I don't think this is possible due to the previous check,
+		// but I'm not confident enough in that assessment to make this a panic.
```

Comment:
We could ditch this possibly. This seems more like something for reviewers to see.

Reply:
I looked more closely and figured out that the check could just be removed 👍